Framework/DataHandling/inc/MantidDataHandling/LoadBankFromDiskTask.h

```diff
@@ -54,12 +54,13 @@ class MANTID_DATAHANDLING_DLL LoadBankFromDiskTask : public Kernel::Task {
 
 private:
   void loadPulseTimes(::NeXus::File &file);
-  void loadEventIndex(::NeXus::File &file, std::vector<uint64_t> &event_index);
+  std::unique_ptr<std::vector<uint64_t>> loadEventIndex(::NeXus::File &file);
```

Comment:
I think it is more idiomatic to return a `std::vector<uint64_t>`, then at the end use `std::move` to move contents of the container into a `boost::shared_ptr`.

Reply:
Good point. That is clearer.